### PR TITLE
Adds ability to login with email when specifying it

### PR DESCRIPTION
**Pull request by flovilmart**

### CODE REVIEW

File: spec/ParseUser.spec.js [hunk 1]
```diff
@@ -3200,4 +3200,147 @@ describe('Parse.User testing', () => {
...
+        headers: {
+          'X-Parse-Application-Id': Parse.applicationId,
+          'X-Parse-REST-API-Key': 'rest',
+        },
+      }
+      return rp.get(options);
+    }).then(done).catch(done.fail);
+  });
+
+  it('can to login when both email and username are passed', (done) => {
```
Comment (montymxb) [inline spec/ParseUser.spec.js]: `can to`, probably should just `can`

File: spec/ParseUser.spec.js [hunk 2]
```diff
@@ -3200,4 +3200,147 @@ describe('Parse.User testing', () => {
...
+        headers: {
+          'X-Parse-Application-Id': Parse.applicationId,
+          'X-Parse-REST-API-Key': 'rest',
+        },
+      }
+      return rp.get(options);
+    }).then(done).catch(done.fail);
+  });
+
+  it('fails to login when username dont match email', (done) => {
```
Comment (montymxb) [inline spec/ParseUser.spec.js]: not critical but `dont` is bugging me

File: spec/ParseUser.spec.js [hunk 3]
```diff
@@ -3200,4 +3200,147 @@ describe('Parse.User testing', () => {
...
+        headers: {
+          'X-Parse-Application-Id': Parse.applicationId,
+          'X-Parse-REST-API-Key': 'rest',
+        },
+        json: { email: "yo@lo.com", password: 'yolopass'}
+      }
+      return rp.get(options);
+    }).then(done).catch(done.fail);
+  });
+
```
Comment (montymxb) [inline spec/ParseUser.spec.js]: Could have an additional test here for trying `cannot login with email and bad password`, just to cover if a possible bug were introduced regarding password validation with an email at any point.

File: spec/ParseUser.spec.js (hunk 1 above)
Comment (flovilmart) [reply spec/ParseUser.spec.js]: yep!

File: spec/ParseUser.spec.js (hunk 3 above)
Comment (flovilmart) [reply spec/ParseUser.spec.js]: Added the test on the next line.

**Comment by flovilmart** [comment]
Thanks man!

**Pull request merged**